src/instructlab/rag/haystack/components/document_splitter.py

```diff
@@ -0,0 +1,102 @@
+# Standard
+from typing import Any, Dict, List, cast
+import logging
+
+# Third Party
+from docling_core.transforms.chunker.hybrid_chunker import HybridChunker
+from docling_core.types import DoclingDocument
+from docling_core.types.legacy_doc.document import (
+    ExportedCCSDocument as LegacyDoclingDocument,
+)
+from docling_core.utils.legacy import legacy_to_docling_document
+from haystack import Document, component  # type: ignore
+from haystack.core.serialization import (  # type: ignore
+    default_from_dict,
+    default_to_dict,
+)
+from pydantic_core._pydantic_core import ValidationError
+
+# First Party
+from instructlab.defaults import DEFAULTS
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class DoclingDocumentSplitter:
```

Comment:
Can we add unit tests for this class and its functions?

Reply:
Sure, I can add more tests and increment the coverage which is quite low:
```
src/instructlab/rag/haystack/components/document_splitter.py      52     32    38%
```

I just want to anticipate you that sooner or later this implementation will be replace by the equivalent one provided by the `docling-haystack` package. See this [issue #4](https://github.com/DS4SD/docling-haystack/issues/4)